diff --git a/include/config_distro_defaults.h b/include/config_distro_defaults.h
index 7532f22..5d18a4b 100644
--- a/include/config_distro_defaults.h
+++ b/include/config_distro_defaults.h
@@ -45,7 +45,7 @@
 
 #define CONFIG_CMDLINE_EDITING
 #define CONFIG_AUTO_COMPLETE
-#define CONFIG_BOOTDELAY     0
+#define CONFIG_BOOTDELAY     2
 #define CONFIG_SYS_LONGHELP
 #define CONFIG_MENU
 #define CONFIG_DOS_PARTITION
diff --git a/include/configs/sunxi-common.h b/include/configs/sunxi-common.h
index 0ff09ca..019a1bb 100644
--- a/include/configs/sunxi-common.h
+++ b/include/configs/sunxi-common.h
@@ -137,8 +137,6 @@
 #define RUN_BOOT_RAM	""
 #endif
 
-
-
 #define CONFIG_BOOTCOMMAND \
 	RUN_BOOT_RAM \
 	"if run loadbootenv; then " \
@@ -168,7 +166,6 @@
 	"panicarg=panic=10\0" \
 	"extraargs=\0" \
 	"loglevel=8\0" \
-	"bootdelay=0\0" \
 	"scriptaddr=0x44000000\0" \
 	"device=mmc\0" \
 	"partition=0:1\0" \
@@ -188,26 +185,26 @@
 	"bootscr=boot.scr\0" \
 	"script=script.bin\0" \
 	"loadbootscr=" \
-	  "ext4load $device $partition $scriptaddr ${bootscr}" \
+	  "fatload $device $partition $scriptaddr ${bootscr}" \
 	  " || " \
-	  "ext4load $device $partition $scriptaddr boot/${bootscr}" \
+	  "ext2load $device $partition $scriptaddr boot/${bootscr}" \
 	  " ||" \
-	  "fatload $device $partition $scriptaddr ${bootscr}" \
+	  "ext2load $device $partition $scriptaddr ${bootscr}" \
 	  "\0" \
 	"loadbootenv=" \
-	  "ext4load $device $partition $scriptaddr ${bootenv}" \
+	  "fatload $device $partition $scriptaddr ${bootenv}" \
 	  " || " \
-	  "ext4load $device $partition $scriptaddr boot/${bootenv}" \
+	  "ext2load $device $partition $scriptaddr boot/${bootenv}" \
 	  " || " \
-	  "fatload $device $partition $scriptaddr ${bootenv}" \
+	  "ext2load $device $partition $scriptaddr ${bootenv}" \
 	  "\0" \
 	"loadkernel=" \
 	  "if "\
 	    "bootpath=/boot/" \
 	    " && " \
-	    "ext4load $device $partition 0x43000000 ${bootpath}${script}" \
+	    "ext2load $device $partition 0x43000000 ${bootpath}${script}" \
 	    " && " \
-	    "ext4load $device $partition 0x48000000 ${bootpath}${kernel}" \
+	    "ext2load $device $partition 0x48000000 ${bootpath}${kernel}" \
 	  ";then true; elif " \
 	    "bootpath=/" \
 	    " && " \
@@ -217,9 +214,9 @@
 	  ";then true; elif " \
 	    "bootpath=/" \
 	    " && " \
-	    "ext4load $device $partition 0x43000000 ${bootpath}${script}" \
+	    "ext2load $device $partition 0x43000000 ${bootpath}${script}" \
 	    " && " \
-	    "ext4load $device $partition 0x48000000 ${bootpath}${kernel}" \
+	    "ext2load $device $partition 0x48000000 ${bootpath}${kernel}" \
 	  ";then true; else "\
 	    "false" \
 	  ";fi" \
@@ -426,8 +423,5 @@
 #ifndef CONFIG_SPL_BUILD
 #include <config_distro_defaults.h>
 #endif
-#ifdef CONFIG_BOOTDELAY
-#undef CONFIG_BOOTDELAY
-#define CONFIG_BOOTDELAY 0
-#endif 
+
 #endif /* _SUNXI_COMMON_CONFIG_H */
